typed-racket-lib/typed-racket/utils/opaque-object.rkt

```diff
@@ -198,5 +206,9 @@
         #:property prop:flat-contract
         (build-flat-contract-property
          #:name restrict-typed-field-name
-         #:stronger equal?
+         #:stronger
+         (λ (this that)
+           (define this-name (restrict-typed-field/c-name this))
```

Comment:
What sort of data do these accessors produce?

Reply:
I think they should always be symbols, the code building them is at https://github.com/dfeltey/typed-racket/blob/2eeed5eb40ce76e0b97a74153d26efee57cd24af/typed-racket-lib/typed-racket/utils/opaque-object.rkt#L67

Reply:
Can you give an example of one of these names? Is it just the field name (I assume it's something more complicated)?

Reply:
I think it's just the field or method name, for example `'guts ` in the example.

Reply:
But then won't this fire for two contracts that just agree on the method name? Maybe I'm missing something important about how this works.

Reply:
It would fire for 2 contracts that enforce there is a method of that name, but that it can never be called. There's no associated contract because there's no type given for the method and TR has to guard it to make sure it can't be called.

Reply:
Ah, now I understand. Just to make sure:

This changes `contract-stronger?` to recognize that if two contracts both restrict the same particular field to not be callable, then they are equally-strong.

If that's right, can you put it in the commit message?

Reply:
Yes that's right, if things look good with the new test case I'll fixup the commit message when I squash and push.

Reply:
Great. This all looks good -- let's make sure Travis CI is happy and then merge.